webapp/channels/src/components/emoji/emoji_list_item/emoji_list_item.tsx

```diff
@@ -22,72 +22,69 @@ export type Props = {
         }
-
-        this.props.actions.deleteCustomEmoji(this.props.emoji.id);
+        deleteCustomEmoji(emoji.id);
     };
 
-    render(): JSX.Element {
-        const emoji = this.props.emoji as CustomEmoji;
-        const creatorUsername = this.props.creatorUsername;
-        let creatorDisplayName = this.props.creatorDisplayName;
-
-        if (creatorUsername && creatorUsername !== creatorDisplayName) {
-            creatorDisplayName += ' (@' + creatorUsername + ')';
-        }
+    if (creatorUsername && creatorUsername !== creatorDisplayName) {
+        // eslint-disable-next-line no-param-reassign
+        creatorDisplayName += ' (@' + creatorUsername + ')';
+    }
 
-        let deleteButton = <DeleteEmojiButton onDelete={this.handleDelete}/>;
+    let deleteButton = <DeleteEmojiButton onDelete={handleDelete}/>;
 
-        if (emoji.creator_id === this.props.currentUserId) {
-            deleteButton = (
-                <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
+    if (emoji.creator_id === currentUserId) {
+        deleteButton = (
+            <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
+                {deleteButton}
+            </AnyTeamPermissionGate>
+        );
+    } else {
+        deleteButton = (
+            <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
+                <AnyTeamPermissionGate permissions={[Permissions.DELETE_OTHERS_EMOJIS]}>
                     {deleteButton}
                 </AnyTeamPermissionGate>
-            );
-        } else {
-            deleteButton = (
-                <AnyTeamPermissionGate permissions={[Permissions.DELETE_EMOJIS]}>
-                    <AnyTeamPermissionGate permissions={[Permissions.DELETE_OTHERS_EMOJIS]}>
-                        {deleteButton}
-                    </AnyTeamPermissionGate>
-                </AnyTeamPermissionGate>
-            );
-        }
-
-        return (
-            <tr className='backstage-list__item'>
-                <td className='emoji-list__name'>
-                    {':' + emoji.name + ':'}
-                </td>
-                <td className='emoji-list__image'>
-                    <span
-                        className='emoticon'
-                        style={{backgroundImage: 'url(' + Client4.getCustomEmojiImageUrl(emoji.id) + ')'}}
-                    />
-                </td>
-                <td className='emoji-list__creator'>
-                    {creatorDisplayName}
-                </td>
-                <td className='emoji-list-item_actions'>
-                    {deleteButton}
-                </td>
-            </tr>
+            </AnyTeamPermissionGate>
         );
     }
-}
+
+    return (
+        <tr className='backstage-list__item'>
+            <td className='emoji-list__name'>{':' + emoji.name + ':'}</td>
+            <td className='emoji-list__image'>
+                <span
+                    className='emoticon'
+                    style={{
```

Comment:
It doesn't make much of a difference here since it is a basic html element, but just to be purist, can you move this style to a memo, so it doesn't create a new object on every render?